File: 643-maximum-average-subarray-i/solution.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        r_total = 0
        a = 0
        for idx, i in enumerate(nums):
            if idx < k:
                r_total += nums[idx]
                a = r_total/k
            else:
                r_total -= nums[idx-k]
                r_total += i
                if r_total/k > a:
                    a = r_total/k
        return a
        

File: 643-maximum-average-subarray-i/solution.go
package main

func findMaxAverage(nums []int, k int) float64 {
	var r_total float64 = 0.0
	var a float64 = 0.0
	for idx, i := range nums {
		if idx < k {
			r_total = r_total + float64(i)
			a = r_total / float64(k)
		} else {
			r_total = r_total - float64(nums[idx-k]) + float64(i)
			if r_total/float64(k) > a {
				a = r_total / float64(k)
			}
		}
	}
	return a

}
